File: python-sidecar/office_claw_sidecar/local_stack/presets.py
"""
로컬 스택 프리셋 — JS `src/lib/localStack`와 동일한 기본값 유지.

단 `planner_model`은 JS 쪽에 대응물을 두지 않는다. 이 태그는 레지스트리에서
pull할 수 있는 모델이 아니라 로컬에서 직접 만든 파인튜닝 산출물이라
(`deploy/ollama/Modelfile.ax7b-planner-v2`), 설치 마법사가 다룰 수 있는 대상이
아니다. 플래너 경로는 sidecar 안에서만 돌기 때문에 여기서만 알면 된다.
"""

from __future__ import annotations

from typing import TypedDict


class LlmConfig(TypedDict):
    provider: str
    model: str
    planner_model: str


class LocalStackPreset(TypedDict):
    id: str
    label: str
    hf_model: str
    ollama_model: str
    multimodal: bool
    llm: LlmConfig


QWEN3_OPENCLAW: LocalStackPreset = {
    "id": "qwen3-openclaw",
    "label": "A.Dot 4.0 Light + OpenClaw (로컬)",
    "hf_model": "skt/A.X-4.0-Light",
    "ollama_model": "skt/A.X-4.0-Light:latest",
    "multimodal": False,
    # model은 범용 대화용, planner_model은 Excel Live 계획 수립 전용이다.
    # 파인튜닝본은 계획 JSON만 뱉도록 학습돼 일반 대화에는 쓸 수 없으므로 분리한다.
    # held-out 18건 체크포인트 비교: v3 최종본 seq 일치 67% (v2: 56% at ckpt-125, 50% at ckpt-250).
    "llm": {
        "provider": "ollama",
        "model": "skt/A.X-4.0-Light:latest",
        "planner_model": "ax7bplanner-v3:latest",
    },
}

# 이전 코드/테스트 호환용 별칭
QWEN3_LOCAL = QWEN3_OPENCLAW

PRESETS: dict[str, LocalStackPreset] = {
    QWEN3_OPENCLAW["id"]: QWEN3_OPENCLAW,
}

DEFAULT_PRESET_ID = QWEN3_OPENCLAW["id"]


def get_preset(preset_id: str = DEFAULT_PRESET_ID) -> LocalStackPreset:
    preset = PRESETS.get(preset_id)
    if preset is None:
        raise KeyError(f"unknown local stack preset: {preset_id}")
    return preset


def get_default_llm_config() -> LlmConfig:
    return dict(get_preset()["llm"])

File: apps/desktop/src/lib/localStack/presets.js
/**
 * 로컬 AI 스택 프리셋 — Ollama 모델 + OpenClaw 게이트웨이 조합.
 */

import { QWEN3_OPENCLAW_PRESET } from "./qwen3OpenClaw.js";

/** @typedef {typeof QWEN3_OPENCLAW_PRESET} LocalStackPreset */

export const LOCAL_STACK_PRESETS = Object.freeze({
  [QWEN3_OPENCLAW_PRESET.id]: QWEN3_OPENCLAW_PRESET,
});

/** 앱 기본 로컬 스택 */
export const DEFAULT_PRESET_ID = QWEN3_OPENCLAW_PRESET.id;

/**
 * @param {string} [presetId]
 * @returns {LocalStackPreset}
 */
export function getLocalStackPreset(presetId = DEFAULT_PRESET_ID) {
  const preset = LOCAL_STACK_PRESETS[presetId];
  if (!preset) {
    throw new Error(`알 수 없는 로컬 스택 프리셋: ${presetId}`);
  }
  return preset;
}

/** LocalAISetupWizard 모델 선택 목록 — 첫 항목이 기본 프리셋 */
export const LOCAL_STACK_MODEL_OPTIONS = Object.freeze([
  {
    id: QWEN3_OPENCLAW_PRESET.ollamaModel,
    label: "skt/A.X-4.0-Light:latest (권장, ~4~6GB)",
    note: "에이닷 경량 계열 · 한국어 자연어 명령 안정성 우선",
    presetId: QWEN3_OPENCLAW_PRESET.id,
  },
]);
// 2026-09-06: `qwen3:8b` 선택지를 뺐다. 앱은 대화 A.X-4.0-Light + 계획 ax7bplanner-v3 둘로만 검증됐고,
// 사용자가 다른 모델을 고르면 플래너와 어긋난 채 "조용히 나빠지는" 상태가 된다.

export const DEFAULT_OLLAMA_MODEL = QWEN3_OPENCLAW_PRESET.ollamaModel;

/**
 * Excel 계획 수립 전용 플래너(사이드카 local_stack/presets.py 의 planner_model 과 같은 이름).
 * 마법사가 범용 모델만 받고 이 모델을 안 받아 새 PC 에서 "플래너가 조용히 죽는" 상태로
 * 시작했다(2026-09-06 실측). 범용 모델과 함께 받는다.
 */
export const DEFAULT_PLANNER_MODEL = "ax7bplanner-v3:latest";
